### ISSUE TITLE
fortios_systemreplacemsg_admin - object already exists

### ISSUE THREAD

**Issue by so-jelly** [fortinetdev/terraform-provider-fortios#139]
```
resource "fortios_systemreplacemsg_admin" "admin" {
  msg_type = "pre_admin-disclaimer-text"
  buffer   = "beware"
  header   = "none"
  format   = "text"
}
```

Unable to 'create' this resources, although it can be imported.
[httpsd 1789 - 1613223127    error] cmdb_commit_from_json[1484] -- object already exists (mkey='pre_admin-disclaimer-text')

**Comment by frankshen01** [comment]
Hi @so-jelly, thank you for raising this issue! I just checked the resource and realized that the predefined resource can only be modified, cannot be deleted/created. Therefore, the current method of processing creation of this resource is wrong. The bug will be fixed in the next release(within about 2 weeks).  Can you please use the following workaround first? It can play a complete and correct effect for the feature.

**validation:**
```
# cat main.tf
provider "fortios" {
  hostname = "192.168.52.177"
  insecure = "true"
  token    = "GNH7r40H65GNb46kd4rG8rtrmn0fr1"
}

variable msg {
  type        = string
  default     = "beware"
}

resource "fortios_json_generic_api" "test2" {
  path   = "/api/v2/cmdb/system.replacemsg/admin/pre_admin-disclaimer-text"
  method = "PUT"
  json   = <<EOF
{
  "msg_type": "pre_admin-disclaimer-text",
  "buffer": "${var.msg}",
  "header": "none",
  "format": "text"
}
EOF
}
```
```
# terraform apply
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # fortios_json_generic_api.test2 will be created
  + resource "fortios_json_generic_api" "test2" {
      + id       = (known after apply)
      + json     = jsonencode(
            {
              + buffer   = "beware"
              + format   = "text"
              + header   = "none"
              + msg_type = "pre_admin-disclaimer-text"
            }
        )
      + method   = "PUT"
      + path     = "/api/v2/cmdb/system.replacemsg/admin/pre_admin-disclaimer-text"
      + response = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

fortios_json_generic_api.test2: Creating...
fortios_json_generic_api.test2: Creation complete after 0s [id=JsonGenericApib1a695c2-bd12-44ec-ae34-94afa2f6c8cd]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: terraform.tfstate
```
```
# terraform plan
fortios_json_generic_api.test2: Refreshing state... [id=JsonGenericApib1a695c2-bd12-44ec-ae34-94afa2f6c8cd]

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```
```
# terraform destroy

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # fortios_json_generic_api.test2 will be destroyed
  - resource "fortios_json_generic_api" "test2" {
      - id       = "JsonGenericApib1a695c2-bd12-44ec-ae34-94afa2f6c8cd" -> null
      - json     = jsonencode(
            {
              - buffer   = "beware"
              - format   = "text"
              - header   = "none"
              - msg_type = "pre_admin-disclaimer-text"
            }
        ) -> null
      - method   = "PUT" -> null
      - path     = "/api/v2/cmdb/system.replacemsg/admin/pre_admin-disclaimer-text" -> null
      - response = jsonencode(
            {
              - build            = 1723
              - http_method      = "PUT"
              - http_status      = 200
              - mkey             = "pre_admin-disclaimer-text"
              - name             = "admin"
              - path             = "system.replacemsg"
              - revision         = "e96755d89fb3f9c0cb746d6302543a5d"
              - revision_changed = false
              - serial           = "FGVM02TM20009055"
              - status           = "success"
              - vdom             = "root"
              - version          = "v6.4.2"
            }
        ) -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

fortios_json_generic_api.test2: Destroying... [id=JsonGenericApib1a695c2-bd12-44ec-ae34-94afa2f6c8cd]
fortios_json_generic_api.test2: Destruction complete after 0s

Destroy complete! Resources: 1 destroyed.
```
Thanks!

**Comment by so-jelly** [comment]
This works, as a further note requires double escaping new lines
"line1\nline2" should use "line1\\nline2"

**Comment by frankshen01** [comment]
Yes, this is indeed part of the Terraform Language syntax, see the `Escape Sequences` section in:
https://www.terraform.io/docs/language/expressions/strings.html

**Comment by frankshen01** [comment]
@so-jelly  Thanks for your improvement! This has been fixed and released in version 1.10.3 of the Terraform FortiOS provider. Please see: https://registry.terraform.io/providers/fortinetdev/fortios/latest and https://registry.terraform.io/providers/fortinetdev/fortios/latest/docs/resources/fortios_systemreplacemsg_admin
 for details. 

**Examples:** 

```
# cat main.tf
provider "fortios" {
  hostname = "192.168.52.177"
  insecure = "true"
  token    = "GNH7r40H65GNb46kd4rG8rtrmn0fr1"
}

resource "fortios_systemreplacemsg_admin" "trname" {
  buffer = chomp(
    <<-EOT
    POST WARNING:
    This is a private computer system. Unauthorized access or use
    is prohibited and subject to prosecution and/or disciplinary
    action. Any use of this system constitutes consent to
    monitoring at all times and users are not entitled to any
    expectation of privacy. If monitoring reveals possible evidence
    of violation of criminal statutes, this evidence and any other
    related information, including identification information about
    the user, may be provided to law enforcement officials.
    If monitoring reveals violations of security regulations or
    unauthorized use, employees who violate security regulations or
    make unauthorized use of this system are subject to appropriate
    disciplinary action.

    %%LAST_SUCCESSFUL_LOGIN%%
    %%LAST_FAILED_LOGIN%%
  EOT
  )
  format   = "text"
  header   = "none"
  msg_type = "post_admin-disclaimer-text"
}
```
```
# terraform plan
.....

# terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # fortios_systemreplacemsg_admin.trname will be created
  + resource "fortios_systemreplacemsg_admin" "trname" {
      + buffer   = <<~EOT
            POST WARNING:
            This is a private computer system. Unauthorized access or use
            is prohibited and subject to prosecution and/or disciplinary
            action. Any use of this system constitutes consent to
            monitoring at all times and users are not entitled to any
            expectation of privacy. If monitoring reveals possible evidence
            of violation of criminal statutes, this evidence and any other
            related information, including identification information about
            the user, may be provided to law enforcement officials.
            If monitoring reveals violations of security regulations or
            unauthorized use, employees who violate security regulations or
            make unauthorized use of this system are subject to appropriate
            disciplinary action.

            %%LAST_SUCCESSFUL_LOGIN%%
            %%LAST_FAILED_LOGIN%%
        EOT
      + format   = "text"
      + header   = "none"
      + id       = (known after apply)
      + msg_type = "post_admin-disclaimer-text"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

fortios_systemreplacemsg_admin.trname: Creating...
fortios_systemreplacemsg_admin.trname: Creation complete after 0s [id=post_admin-disclaimer-text]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

```
labfirewall # config system replacemsg admin post_admin-disclaimer-text

labfirewall (post_admin-discl~ext) # show full-configuration
config system replacemsg admin "post_admin-disclaimer-text"
    set buffer "POST WARNING:
This is a private computer system. Unauthorized access or use
is prohibited and subject to prosecution and/or disciplinary
action. Any use of this system constitutes consent to
monitoring at all times and users are not entitled to any
expectation of privacy. If monitoring reveals possible evidence
of violation of criminal statutes, this evidence and any other
related information, including identification information about
the user, may be provided to law enforcement officials.
If monitoring reveals violations of security regulations or
unauthorized use, employees who violate security regulations or
make unauthorized use of this system are subject to appropriate
disciplinary action.

%%LAST_SUCCESSFUL_LOGIN%%
%%LAST_FAILED_LOGIN%%"
    set header none
    set format text
end
```

```
# terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

fortios_systemreplacemsg_admin.trname: Refreshing state... [id=post_admin-disclaimer-text]

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.
```

```
# terraform destroy
fortios_systemreplacemsg_admin.trname: Refreshing state... [id=post_admin-disclaimer-text]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # fortios_systemreplacemsg_admin.trname will be destroyed
  - resource "fortios_systemreplacemsg_admin" "trname" {
      - buffer   = <<~EOT
            POST WARNING:
            This is a private computer system. Unauthorized access or use
            is prohibited and subject to prosecution and/or disciplinary
            action. Any use of this system constitutes consent to
            monitoring at all times and users are not entitled to any
            expectation of privacy. If monitoring reveals possible evidence
            of violation of criminal statutes, this evidence and any other
            related information, including identification information about
            the user, may be provided to law enforcement officials.
            If monitoring reveals violations of security regulations or
            unauthorized use, employees who violate security regulations or
            make unauthorized use of this system are subject to appropriate
            disciplinary action.

            %%LAST_SUCCESSFUL_LOGIN%%
            %%LAST_FAILED_LOGIN%%
        EOT -> null
      - format   = "text" -> null
      - header   = "none" -> null
      - id       = "post_admin-disclaimer-text" -> null
      - msg_type = "post_admin-disclaimer-text" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

fortios_systemreplacemsg_admin.trname: Destroying... [id=post_admin-disclaimer-text]
fortios_systemreplacemsg_admin.trname: Destruction complete after 0s

Destroy complete! Resources: 1 destroyed.
```

Thanks!
